dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/condition/config/model/ConditionRuleParser.java

```diff
@@ -40,14 +47,35 @@
  */
 public class ConditionRuleParser {
 
-    public static ConditionRouterRule parse(String rawRule) {
+    private static final ErrorTypeAwareLogger logger = LoggerFactory.getErrorTypeAwareLogger(ConditionRuleParser.class);
+
+    public static AbstractRouterRule parse(String rawRule) {
+        AbstractRouterRule rule;
         Yaml yaml = new Yaml(new SafeConstructor(new LoaderOptions()));
         Map<String, Object> map = yaml.load(rawRule);
-        ConditionRouterRule rule = ConditionRouterRule.parseFromMap(map);
-        rule.setRawRule(rawRule);
-        if (CollectionUtils.isEmpty(rule.getConditions())) {
+        String confVersion = (String) map.get(CONFIG_VERSION_KEY);
+
+        if (confVersion != null && confVersion.toLowerCase().startsWith(RULE_VERSION_V31)) {
+            rule = MultiDestConditionRouterRule.parseFromMap(map);
+            if (CollectionUtils.isEmpty(((MultiDestConditionRouterRule) rule).getConditions())) {
+                rule.setValid(false);
+            }
+        } else if (confVersion != null && confVersion.compareToIgnoreCase(RULE_VERSION_V31) > 0) {
+            logger.warn(
+                    CLUSTER_FAILED_RULE_PARSING,
+                    "Invalid condition config version number.",
+                    "",
+                    "Ignore this configuration. Only " + RULE_VERSION_V31 + " and below are supported in this release");
+            rule = ConditionRouterRule.parseFromMap(map);
```

Comment:
Should we continue with parsing when found configVersion not match?

Reply:
An alternative is to simply create a new ConditionRouterRule object.Otherwise, the rule property is null. Even though there may be a null test, I think it's best to try not to generate null.

Reply:
Just set it to null and add a unit test